### ISSUE TITLE
Pagination by date

### ISSUE THREAD

**Issue by est** [getpelican/pelican#3058]
- [X] I have searched the [issues](https://github.com/getpelican/pelican/issues?q=is%3Aissue) (including closed ones) and believe that this is not a duplicate.
- [X] I have searched the [documentation](https://docs.getpelican.com/) and believe that my question is not covered.
- [X] I am willing to lend a hand to help implement this feature. 

## Feature Request

Is it possible to add pagination by date? Instead of fixed number of articles per page, place like weekly article per page, or on month per page. 

In this way, static files are better managed because old page's content is fixed and rarely change. You don't add one new article and have to update all pagination pages.

**Comment by avaris** [comment]
Unfortunately, no, that's not supported. And the way things are implemented right now, there is not an easy (non-hacky-ish) way to replace the built-in paginator, say, with a plugin.

**Comment by GiovanH** [comment]
Hm. I know it's definitely possible to have traversable index pages like date, using settings like
```python
INDEX_SAVE_AS = 'index.html'
ARTICLE_URL = 'blog/{date:%Y}/{date:%m}/{date:%d}/{slug}/'
ARTICLE_SAVE_AS = 'blog/{date:%Y}/{date:%m}/{date:%d}/{slug}/index.html'

ARCHIVES_URL = 'blog/'
ARCHIVES_SAVE_AS = 'blog/index.html'

YEAR_ARCHIVE_SAVE_AS = 'blog/{date:%Y}/index.html'
MONTH_ARCHIVE_SAVE_AS = 'blog/{date:%Y}/{date:%m}/index.html'
DAY_ARCHIVE_SAVE_AS = 'blog/{date:%Y}/{date:%m}/{date:%d}/index.html'
```
If you're willing to use those pages, this might be doable with just the `neighbors` plugin and some clever templating.

**Comment by GiovanH** [comment]
Yep, here is code that uses period-archives as a month-by-month paginated index using the neighbors plugin:

```j2
    {% set pagination_article = dates[0].prev_article %}
    {% if pagination_article and pagination_article.date %}
        <a class="prev" href="{{ SITEURL }}/{{ MONTH_ARCHIVE_SAVE_AS.format(date=pagination_article.date) }}">
            {{ pagination_article.date.isoformat() }}
        </a>
    {% else %}<span class="prev"></span>{% endif %}

    <!-- <a href="{{ SITEURL }}/{{ ARCHIVES_URL }}">Blog archive</a> -->

    {% set pagination_article = dates[-1].next_article %}
    {% if pagination_article and pagination_article.date %}
        <a class="next" href="{{ SITEURL }}/{{ MONTH_ARCHIVE_SAVE_AS.format(date=pagination_article.date) }}">
            {{ pagination_article.date.isoformat() }} &rarr;
        </a>
    {% else %}<span class="next"></span>{% endif %}
```

**Comment by justinmayer** [comment]
> In this way, static files are more "static" because old page's content is fixed and rarely changes. You don't add one new article and have to update all pagination pages.

If that is the goal, see the `PAGINATION_PATTERNS` enhancement introduced in Pelican 4.6: https://getpelican.com/blog/pelican-4.6-released/

Thanks to @avaris and @GiovanH for the helpful responses! 🏅
